src/Physics/Simulator.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <LinearMath/btVector3.h>
+#include <glm/glm.hpp>
+#include <memory>
+
+class btBroadphaseInterface;
+class btDefaultCollisionConfiguration;
+class btCollisionDispatcher;
+class btSequentialImpulseConstraintSolver;
+class btDiscreteDynamicsWorld;
+class btCollisionShape;
+class btDefaultMotionState;
+class btRigidBody;
+
+namespace Physics {
+
+    class RigidBody;
+
+    /// Simulates interactions among given objects.
+    class Simulator {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') class 'Simulator' does not have a copy constructor which is recommended since the class contains a pointer to allocated memory. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AnoCopyConstructor)